maths/binary_exp_mod.py

```diff
@@ -17,6 +17,74 @@ def bin_exp_mod(a, n, b):
     return (r * r) % b
 
 
+def binary_exponentiation_mod_multiplication(a, b, c):
```

Comment:
This function also appears to calculate (modular) multiplication rather than exponentiation, so I feel like this might also belong better in a separate file for binary multiplication